### ISSUE TITLE
Flow Validity Mask calculation 

### ISSUE THREAD

**Issue by Huihui1002** [MaureenZOU/TSAM#14]
Dear Xueyan,

May I ask how do you calculate the Flow Validity Mask that you described in Section 3.3?
Is this part of code in src/model/tsm/temperal_shift.py?
<img width="1060" alt="validity" src="https://user-images.githubusercontent.com/23012974/130047075-56f8e355-74ac-4a71-b33c-07572385c84b.png">
But there is no cycle consistency here.

And how do you calculate the left_flowmask and right_flowmask?

Another question is:
In line 109, torch.clamp(right_flowmask + (torch.sum(left_feature, dim=2, keepdim=True)==0).float(), 0, 1).int().float().
Why left_feature is used here?

Many thanks for your help in advance!
